src/storage/pooled_storage_manager.h

```diff
@@ -69,6 +70,11 @@ class GPUPooledStorageManager final : public StorageManager {
       LOG(FATAL) << "MXNET_GPU_MEM_POOL_PAGE_SIZE cannot be set to a value smaller than " << NDEV \
                  << ". Got " << page_size_ << ".";
     }
+    memory_limit_percentage_ = dmlc::GetEnv<double>("MXNET_GPU_MEM_LIMIT", 100.0);
+    if (memory_limit_percentage_ <= 0 || memory_limit_percentage_ > 100) {
+      LOG(FATAL) << "Invalid memory limit percentage given: " << memory_limit_percentage_
+                 << std::endl;
+    }
```

Comment:
How is this variable semantically different from the `MXNET_GPU_MEM_POOL_RESERVE` variable?

Reply:
according to https://mxnet-bing.readthedocs.io/en/latest/how_to/env_var.html

> MXNET_GPU_MEM_POOL_RESERVE (default=5) is The percentage of GPU memory to reserve for things other than the GPU array, such as kernel launch or cudnn handle space.

So its more like the inverse. Instead of saying how much you want to "leave for other things", `MXNET_GPU_MEM_LIMIT` is for how much memory you want to "give" MXNet in particular.